io-engine/src/grpc/v1/nexus.rs

```diff
@@ -1147,4 +1148,47 @@ impl NexusRpc for NexusService {
         })
         .await
     }
+
+    async fn list_rebuild_history(
+        &self,
+        request: Request<ListRebuildHistoryRequest>,
+    ) -> GrpcResult<RebuildHistoryMapResponse> {
+        let args = request.into_inner();
+        let count = args.count.unwrap_or_default();
+        let rx = rpc_submit::<_, _, nexus::Error>(async move {
+            let mut history_map: HashMap<String, RebuildHistoryResponse> =
+                HashMap::new();
+            for nexus in nexus::nexus_iter() {
+                let mut history_list: VecDeque<RebuildHistoryRecord> =
+                    VecDeque::new();
+                let nexus_uuid = nexus.uuid().to_string();
+                for record in
+                    nexus_lookup(&nexus_uuid)?.rebuild_history().iter()
```

Comment:
you don't need to lookup a nexus here, as you already have it (from "for nexus in nexus::nexus_iter()")

Reply:
Incorporated